src/js/pages/system/UnitsHealthDetail.js

```diff
@@ -47,6 +55,24 @@ class UnitsHealthDetail extends mixin(StoreMixin) {
     UnitHealthStore.fetchUnitNodes(this.props.params.unitID);
   }
 
+  onUnitHealthStoreUnitSuccess() {
+    let isLoading = this.state.isLoading & ~FLAGS.UNIT_LOADED;
```

Comment:
+1 Can't agree more. Additionally for the specific case of this PR we could just hide the logical checks away inside of a simple function and be done with it.